baseplate/clients/thrift.py

```diff
@@ -213,38 +218,45 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
                         prot.trans.set_header(b"Edge-Request", edge_context)
 
                     result = method(*args, **kwargs)
-            except TTransportException as exc:
-                # the connection failed for some reason, retry if able
-                span.finish(exc_info=sys.exc_info())
-                last_error = str(exc)
-                if exc.inner is not None:
-                    last_error += f" ({exc.inner})"
-                continue
-            except (TApplicationException, TProtocolException):
-                # these are subclasses of TException but aren't ones that
-                # should be expected in the protocol. this is an error!
-                span.finish(exc_info=sys.exc_info())
-                raise
-            except Error as exc:
-                # a 5xx error is an unexpected exception but not 5xx are
-                # not.
-                if 500 <= exc.code < 600:
+                except TTransportException as exc:
+                    # the connection failed for some reason, retry if able
+                    span.finish(exc_info=sys.exc_info())
+                    last_error = str(exc)
+                    if exc.inner is not None:
+                        last_error += f" ({exc.inner})"
+                    continue
+                except (TApplicationException, TProtocolException):
+                    # these are subclasses of TException but aren't ones that
+                    # should be expected in the protocol. this is an error!
                     span.finish(exc_info=sys.exc_info())
+                    raise
+                except Error as exc:
+                    # a 5xx error is an unexpected exception but not 5xx are
+                    # not.
+                    if 500 <= exc.code < 600:
+                        span.finish(exc_info=sys.exc_info())
+                    else:
+                        span.finish()
+                    raise
+                except TException:
+                    # this is an expected exception, as defined in the IDL
+                    span.finish()
+                    raise
+                except:  # noqa: E722
+                    # something unexpected happened
+                    try:
+                        span.finish(exc_info=sys.exc_info())
+                    except ParentSpanAlreadyFinishedError:
+                        logger.warn("Attempted to finish a span where the parent was already closed")
+                        raise
+                    except:
```

Comment:
One of the errors was trying to reference `span` before it was defined, I figured this will be sure anything bubbles up, but I added explicit handling for the case that triggered this change so I'd be fine with dropping this block.

Reply:
You can safely remove this, since not catching it means raising it.